lib/Dialect/FIRRTL/InstanceGraph.cpp

```diff
@@ -89,3 +90,56 @@ InstanceGraphNode *InstanceGraph::getOrAddNode(StringRef name) {
 Operation *InstanceGraph::getReferencedModule(InstanceOp op) {
   return lookup(op.moduleName())->getModule();
 }
+
+InstanceGraphNode *InstanceGraph::getLCA(InstanceGraphNode *node1,
+                                         InstanceGraphNode *node2) {
+  // Algorithm:
+  // 1. Record all ancestors of node1 in set node1Ancestors.
+  // 2. Visit all ancestors of node2.
+  //  2.1. If node2 ancestor also exists in node1Ancestors set
+  //    2.1.1 And its depth is largest of the common ancestors, then
+  //      2.1.1.1 This is a candidate LCA.
+
+  // Returns nullptr, if the two nodes donot have any common ancestor.
+  InstanceGraphNode *lcaNode = nullptr;
```

Comment:
If you have the depths, you shoudl be able to use an algorithm like this:

while (node1->depth > node2->depth). // and not null
  node1 = node1->getParent()

while (node2->depth > node1->depth)
  node2 = node2->getParent()

// they have the same depth now.
while (node1 != node2 && node1 != null && node2 != null) {
  node1 = node1->getParent()
  node2 = node2->getParent()
}

return node1;